core/src/main/java/com/javadiscord/jdi/core/Discord.java

```diff
@@ -160,6 +160,8 @@ public void stop() {
             this.webSocketManager.stop();
         }
 
+        discordRequestDispatcher.stop();
+
         EXECUTOR.shutdown();
         try {
             if (!EXECUTOR.awaitTermination(30, TimeUnit.SECONDS)) {
```

Comment:
Do we really want 30 seconds await time?

Reply:
This is reasonable - I am okay with 30 seconds

Reply:
yea true, misunderstood its implementation